app/org/sagebionetworks/bridge/dynamodb/DynamoScheduledActivityDao.java

```diff
@@ -62,43 +57,24 @@ public ScheduledActivity getActivity(String healthCode, String guid) {
     
     /** {@inheritDoc} */
     @Override
-    public List<ScheduledActivity> getActivities(ScheduleContext context) {
-        DynamoScheduledActivity hashKey = new DynamoScheduledActivity();
-        hashKey.setHealthCode(context.getHealthCode());
-
-        // Exclude everything hidden before *now*
-        AttributeValue attribute = new AttributeValue().withN(Long.toString(context.getNow().getMillis()));
-        Condition condition = new Condition()
-            .withComparisonOperator(ComparisonOperator.GT)
-            .withAttributeValueList(attribute);
-
-        DynamoDBQueryExpression<DynamoScheduledActivity> query = new DynamoDBQueryExpression<DynamoScheduledActivity>()
-            .withQueryFilterEntry("hidesOn", condition)
-            .withHashKeyValues(hashKey);
-
-        PaginatedQueryList<DynamoScheduledActivity> queryResults = mapper.query(DynamoScheduledActivity.class, query);
+    public List<ScheduledActivity> getActivities(DateTimeZone timeZone, List<ScheduledActivity> activities) {
+        if (activities.isEmpty()) {
+            return ImmutableList.of();
+        }
+        List<Object> activitiesToLoad = new ArrayList<Object>(activities);
+        Map<String,List<Object>> resultMap = mapper.batchLoad(activitiesToLoad);
```

Comment:
Because of type erasure, you can always cast a List of ScheduledActivity to a List of Object. No need to create a copy of the list. (We do this in DynamoIndexHelper.)